### ISSUE TITLE
Show an error when resolving shell environment fails or is timing out

### ISSUE THREAD

**Issue by Qythyx** [microsoft/vscode#133910]
Issue Type: <b>Bug</b>

When I launch VSCode on Mac from the Applications folder or via Spotlight Search the path is not correctly set. This means that when I try to run tasks from withing VSCode they fail to find the command. I've confirmed this by creating a test task which  is this:

```json
{
	"version": "2.0.0",
	"tasks": [
		{
			"command": "echo $PATH",
			"type": "shell",
			"label": "test task",
			"problemMatcher": []
		}
	]
}
```

The output of this is `/usr/bin:/bin:/usr/sbin:/sbin`.

If, however, I launch VSCode from the terminal via `open /Applications/Visual\ Studio\ Code.app` then the path is set correctly.

VS Code version: Code 1.60.2 (7f6ab5485bbc008386c4386d08766667e155244e, 2021-09-22T11:59:27.195Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No
Remote OS version: Linux x64 5.11.0-1016-aws
Remote OS version: Linux x64 5.11.0-1016-aws

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz (8 x 2300)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|13, 30, 22|
|Memory (System)|16.00GB (0.92GB free)|
|Process Argv|--crash-reporter-id 0e8bfe3f-3aff-47ae-ae44-87a17d7621cd|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: rainer.cvm.indeed.net|
|OS|Linux x64 5.11.0-1016-aws|
|CPUs|AMD EPYC 7571 (4 x 2199)|
|Memory (System)|15.50GB (6.23GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: rainer.cvm.indeed.net|
|OS|Linux x64 5.11.0-1016-aws|
|CPUs|AMD EPYC 7571 (4 x 2199)|
|Memory (System)|15.50GB (6.23GB free)|
|VM|0%|
</details><details><summary>Extensions (35)</summary>

Extension|Author (truncated)|Version
---|---|---
overtype|ada|0.2.0
npm-intellisense|chr|1.4.0
vscode-markdownlint|Dav|0.44.3
vscode-eslint|dba|2.1.25
gitlens|eam|11.6.0
vscode-npm-script|eg2|0.3.22
prettier-vscode|esb|9.0.0
nasc-touchbar|fel|1.12.0
gitlab-workflow|Git|3.31.0
gc-excelviewer|Gra|3.0.44
vscode-browser-sync|jas|1.3.1
vscode-azureresourcegroups|ms-|0.4.0
vscode-cosmosdb|ms-|0.18.1
vscode-docker|ms-|1.17.0
python|ms-|2021.9.1246542782
vscode-pylance|ms-|2021.9.3
jupyter|ms-|2021.8.2041215044
jupyter-keymap|ms-|1.0.0
remote-ssh|ms-|0.65.8
remote-ssh-edit|ms-|0.65.8
remote-wsl|ms-|0.58.2
vscode-remote-extensionpack|ms-|0.21.0
azure-account|ms-|0.9.8
vscode-jest|Ort|4.1.2
java|red|0.82.0
LiveServer|rit|5.6.1
vs-code-prettier-eslint|rve|3.0.4
vscode-coverage-gutters|rya|2.8.1
html-preview-vscode|tht|0.2.5
vscodeintellicode|Vis|1.2.14
vscode-java-debug|vsc|0.36.0
vscode-java-dependency|vsc|0.18.8
vscode-java-pack|vsc|0.18.5
vscode-java-test|vsc|0.32.0
vscode-maven|vsc|0.34.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
vsccppwt:30364497
pythonvssor306:30344512
bridge0708:30335490
pygetstartedt2:30371810
dockerwalkthru:30370836
bridge0723:30353136
pythonrunftest32:30373476
pythonf5test824:30373475
javagetstartedc:30364665
pythonvspyt187:30373474
pydsgst2:30361792

```

</details>

**Comment by Qythyx** [comment]
Continuing to test this,:

1. Run from Spotlight Search: path problem
2. Run from Finder (double-click VSCode): path ok
3. Run from Spotlight Search: path problem
4. Run from Finder (double-click VSCode): path problem

Again, I'm not doing anything differently when I'm running these attempts, sometimes the path works, sometimes it does not.

My current summary is: running from the Terminal is always ok, running from Finder or Spotlight sometimes works and sometimes has this problem.

**Comment by joaomoreno** [comment]
Do you get any interesting notification when starting VS Code, namely around env and path?

**Comment by Qythyx** [comment]
> Do you get any interesting notification when starting VS Code, namely around env and path?

No, I don't think so. Where would I see such a notification? In the toast in the lower-right where other errors/notifications appear?

**Comment by Qythyx** [comment]
I'm more and more convinced this is somehow timing or resource related. I've done more testing and it seems the problem is more likely to repro when my Mac is busy (ongoing Zoom meeting, opening many tabs in Chrome, etc). I assume this business also slows down launching VSCode, which could aggravate the problem if it is timing related. I don't have any direct evidence of this, just guessing.

For example, I tested it about 8 times without reproing the issue. Then I opened a Chrome window with about 30 tabs. While those were loading I tested again and this time I did repro the issue.

**Comment by Qythyx** [comment]
Ok, I think I know what's happening. I believe it is a timeout. If I add `sleep 10` to my `.zprofile` file then this seems to cause the problem to always happen. But, I'm still not seeing any notification in VS Code. I know in the past I remember seeing some warning about "you shell environment takes to long to start", but I am not seeing that warning now.

**Comment by bpasero** [comment]
See https://github.com/microsoft/vscode/issues/132669 for that. If it takes longer than 10 seconds to resolve your shell environment, VSCode gives up to wait for it. If it takes 10 seconds to load a terminal for you, you should have noticed this even when using terminals normally.

**Comment by Qythyx** [comment]
@bpasero, thanks for the info. I don't actually use a terminal that often. I know it takes a bit of time to start up, but not so much that it was really a problem. Usually when I'm using a terminal it's from within VS Code.

I understand why the notification was removed, but it seems that it might be worth giving some notice to the user that their shell didn't start up correctly. For me I first noticed this problem when I tried to run some commands/tasks from within VS Code. Those failed with "command not found" errors. Debugging that problem to be related to shell startup time has not been easy, so telling the users more directly would be useful.

**Comment by bpasero** [comment]
Thats a fair request. When I removed the notification my hope was that developers would have already noticed that their development setup was broken when opening a terminal takes >10 seconds, but I guess that was a wrong assumption.

**Comment by bpasero** [comment]
**Verification (macOS/Linux):**
* add `sleep 11` to your `.zhsrc` or `.bashrc` file
* open code insiders from dock
* verify you see an error toast that resolving shell env timed out
* remove `sleep 11` again
* configure an environment variable in that same file
* start again from dock
* verify the environment variable is there in your vscode environment

**Comment by andrei-zgirvaci** [comment]
@Qythyx @bpasero you guys rock! 🎉

I have been struggling for a few weeks trying to figure out why my env paths sometimes work, sometimes don't when opening the terminal in vscode.

Thanks to both of you I finally understood that the problem was my config being misconfigured and I managed to fix it!

On average my terminal took 12 seconds to open each time and I lived like that for 2 weeks, now it opens instantly, oh... such a relief! 😅
